services/ingestion/etl/budg/xls/src/events/onParseXLS.js

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+
+import transformRecord from '../lib/transform';
+
+// Destination bucket
+const { BUCKET } = process.env;
+
+export const handler = (event, context, callback) => {
+  /*
+   * Some checks here before going any further
+   */
+
+  // Only work on the first record
+  const snsRecord = event.Records[0];
+
+  // Was the lambda triggered correctly? Is the file extension supported? etc.
+  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
+    return callback('Bad record');
+  }
+
+  /*
+   * Prepare file analysis
+   */
+
+  // Extract message
+  const message = JSON.parse(snsRecord.Sns.Message);
+
+  // Check file extension
+  if (['.xls', '.xlsx'].indexOf(path.extname(message.object.key)) === -1) {
+    return callback('File extension should be .xls or .xlsx');
+  }
+
+  const s3 = new AWS.S3();
+  const XLSX = require('xlsx');
```

Comment:
Imports/requires come first. Please also check https://drone.ne-dev.eu/ec-europa/eubfr-data-lake/206, there is a couple of linting errors out there